### DISCUSSION TITLE
Modal view record rather than edit record

### DISCUSSION THREAD

**Discussion by markdynamo6** [filamentphp/filament#8317] [Help] [answered]
Following this https://filamentphp.com/docs/3.x/panels/resources/viewing-records#viewing-records-in-modals yields the Edit form (albeit in a disabled state) in the modal. Is there a way to use a ViewRecord view. E.g. with an infolist?

**Comment by leandrocfe** [comment]
Would you like to remove edit record? Sorry I didn't understand the issue..

**Comment by markdynamo6** [comment]
No probs. 

I didn't want to remove edit record, I just wanted to have a more designed version of a _view record_ when displayed in a modal. This is possible when using a separate ViewPage, but the docs say:

```If your resource is simple, you may wish to view records in modals rather than on the [View page](https://filamentphp.com/docs/3.x/panels/resources/viewing-records). If this is the case, you can just [delete the view page](https://filamentphp.com/docs/3.x/panels/resources/getting-started#deleting-resource-pages).```

Once the "View page" has been deleted, it's not clear how to then create a custom View Record to go in this modal.

**Comment by markdynamo6** [comment, marked as answer]
Just found the answer:

I just needed to simply add an infolist function into my List Records page.

```php
<?php

namespace App\Filament\Resources\UserResource\Pages;

use App\Filament\Resources\UserResource;
use Filament\Actions;
use Filament\Resources\Pages\ListRecords;
use Filament\Infolists;
use Filament\Infolists\Infolist;

class ListUsers extends ListRecords
{
    protected static string $resource = UserResource::class;

    // This is used to render the modal view record
    public function infolist(Infolist $infolist): Infolist
    {
        return $infolist
            ->schema([
                Infolists\Components\TextEntry::make('name'),
            ]);
    }
}

```

Thanks for replying @leandrocfe
